src/Servers/HttpSys/src/NativeInterop/SafeLibraryHandle.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+using System.Security;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.AspNetCore.Server.HttpSys
+{
+    /// <summary>
+    /// Represents a handle to a Windows module (DLL).
+    /// </summary>
+    internal unsafe sealed class SafeLibraryHandle : SafeHandleZeroOrMinusOneIsInvalid
```

Comment:
What do you think about making this shared source and using it in DataProtection? It does look very low churn, so maybe it's not necessary.

Reply:
You noticed :grin:. I tried briefly but the DataProtection version pulled in additional types. In this copy I was able to trim out unused functionality to avoid pulling in more types.